### PR TITLE
[luci/lang] Add CircleGRU node

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr adds CircleCirGru node in luci/lang

for issue: https://github.com/Samsung/ONE/issues/12320
from draft: https://github.com/Samsung/ONE/pull/12319

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Comment by BalyshevArtem** [comment]
Note: should be rebased after #12597 will be merged

File: compiler/luci/lang/include/luci/IR/CircleNodes.lst [hunk 1]
```diff
@@ -52,6 +52,7 @@ CIRCLE_NODE(GATHER_ND, CircleGatherNd)
 CIRCLE_NODE(GELU, CircleGelu)
 CIRCLE_NODE(GREATER, CircleGreater)
 CIRCLE_NODE(GREATER_EQUAL, CircleGreaterEqual)
+CIRCLE_NODE(CIR_GRU, CircleCirGru)
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/CircleNodes.lst:55]:
```suggestion
// This is ordered as `Gru` not `CirGru`
CIRCLE_NODE(CIR_GRU, CircleCirGru)
```

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Comment by BalyshevArtem** [comment]
@seanshpark, can you please help me to resolve build issue:
```
[2024-02-21T08:25:01.182Z] /opt/jenkins_agent/workspace/nnfw/master/pr-nncc-release@2/compiler/luci/lang/include/luci/IR/CircleNodes.lst: In member function ‘virtual circle::BuiltinOperator {anonymous}::QueryOpCode::visit(const luci::CircleGRU*)’:

[2024-02-21T08:25:01.182Z] /opt/jenkins_agent/workspace/nnfw/master/pr-nncc-release@2/compiler/luci/partition/src/CircleOpCode.cpp:35:62: error: ‘BuiltinOperator_GRU’ was not declared in this scope

[2024-02-21T08:25:01.182Z]    BuiltinOperator visit(const CIRCLE_CLASS *) final { return BuiltinOperator_##OPCODE; }

[2024-02-21T08:25:01.182Z]                                                               ^~~~~~~~~~~~~~~~

[2024-02-21T08:25:01.182Z] /opt/jenkins_agent/workspace/nnfw/master/pr-nncc-release@2/compiler/luci/lang/include/luci/IR/CircleNodes.lst:55:1: note: in expansion of macro ‘CIRCLE_NODE’

[2024-02-21T08:25:01.182Z]  CIRCLE_NODE(GRU, CircleGRU)

[2024-02-21T08:25:01.182Z]  ^~~~~~~~~~~

[2024-02-21T08:25:01.182Z] [ 15%] Building CXX object compiler/luci/pass/CMakeFiles/luci_pass.dir/src/CircleQuantizer.cpp.o

[2024-02-21T08:25:01.441Z] [ 15%] Building CXX object compiler/luci/lang/CMakeFiles/luci_lang_test.dir/src/Nodes/CircleSoftmax.test.cpp.o

[2024-02-21T08:25:01.699Z] [ 15%] Building CXX object compiler/luci-interpreter/src/kernels/CMakeFiles/luci_interpreter_kernels.dir/Relu6.cpp.o

[2024-02-21T08:25:04.215Z] compiler/luci/partition/CMakeFiles/luci_partition.dir/build.make:62: recipe for target 'compiler/luci/partition/CMakeFiles/luci_partition.dir/src/CircleOpCode.cpp.o' failed
```

How can I resolve it? Smth with decencies?

**Comment by seanshpark** [comment]
> How can I resolve it? Smth with decencies?

Ah, we need to add `mio-circle08` and revise to use libs to use `08`
please refer  #12177 and it's links for changes.

**Comment by seanshpark** [comment]
https://github.com/Samsung/ONE/pull/12597#issuecomment-1943011689

adding 08 version was what I was afraid of... lots of changes are needed.
I think this may take some time as lots of guys here including me are mostly out of office in Feb.

**Comment by BalyshevArtem** [comment]
rebased after merging mio-circe 08
@seanshpark, PTAL :)

**Comment by seanshpark** [comment]
> [2024-03-11T09:17:22.287Z] /opt/jenkins_agent/workspace/nnfw/master/pr-nncc-debug/compiler/luci/partition/src/CircleOpCode.cpp:35:62: error: ‘BuiltinOperator_CIR_GRU’ was not declared in this scope
[2024-03-11T09:17:22.287Z]    BuiltinOperator visit(const CIRCLE_CLASS *) final { return BuiltinOperator_##OPCODE; }
[2024-03-11T09:17:22.287Z] /opt/jenkins_agent/workspace/nnfw/master/pr-nncc-debug/compiler/luci/lang/include/luci/IR/CircleNodes.lst:55:1: note: in expansion of macro ‘CIRCLE_NODE’
[2024-03-11T09:17:22.287Z]  CIRCLE_NODE(CIR_GRU, CircleCirGru)

File: compiler/luci/lang/src/Nodes/CircleCirGru.test.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ */
+
+#include "luci/IR/Nodes/CircleCirGru.h"
+
+#include "luci/IR/CircleDialect.h"
+#include "luci/IR/CircleNodeVisitor.h"
+
+#include <gtest/gtest.h>
+
+TEST(CircleCirGruTest, constructor_P)
```
Comment (jinevening) [inline compiler/luci/lang/src/Nodes/CircleCirGru.test.cpp:24]: _P is not necessary

```suggestion
TEST(CircleCirGruTest, constructor)
```

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**